File: dotnet/aspnet-with-ef/HuHao.BlogService.Domain/Core/Exceptions/EntityNotFoundException.cs
using System;

namespace HuHao.BlogService.Domain.Core.Exceptions
{
    public class EntityNotFoundException : DomainException
    {
        public EntityNotFoundException(string message) : base(message)
        {
        }

        public EntityNotFoundException(Type type, Guid id) :
            this($"cannot find the {type.Name.ToLower()} with id {id}")
        {
        }
    }
}

File: java-samples/blog-service/src/main/java/dev/huhao/samples/ddd/blogservice/domain/common/EntityNotFoundException.java
package dev.huhao.samples.ddd.blogservice.domain.common;

import dev.huhao.samples.ddd.blogservice.domain.concepts.Entity;

import java.util.UUID;

public class EntityNotFoundException extends DomainException {
    public EntityNotFoundException(String message) {
        super(message);
    }

    public <T extends Entity> EntityNotFoundException(Class<T> entityClass, UUID id) {
        super("cannot find the " + entityClass.getSimpleName().toLowerCase() + " with id " + id);
    }
}
